internal/monitor/trino.go

```diff
@@ -0,0 +1,224 @@
+package monitor
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/razorpay/trino-gateway/internal/boot"
+	"github.com/razorpay/trino-gateway/internal/provider"
+	"github.com/trinodb/trino-go-client/trino"
+)
+
+type ITrinoClient interface {
```

Comment:
This interface doesn't seem to make a difference to the code. Can we remove it?

Reply:
this will be used in unit tests

since trino is an external service, need an interface to provide mock implementations